dbms/src/Interpreters/HashJoinSpillContext.cpp

```diff
@@ -27,29 +27,37 @@ HashJoinSpillContext::HashJoinSpillContext(
     , max_cached_bytes(std::max(
           build_spill_config.max_cached_data_bytes_in_spiller,
           probe_spill_config.max_cached_data_bytes_in_spiller))
-{
-    /// join does not support auto spill mode
-    auto_spill_mode = false;
-}
+{}
 
 void HashJoinSpillContext::init(size_t partition_num)
 {
     partition_revocable_memories = std::make_unique<std::vector<std::atomic<Int64>>>(partition_num);
     partition_is_spilled = std::make_unique<std::vector<std::atomic<bool>>>(partition_num);
+    partition_spill_status = std::make_unique<std::vector<std::atomic<AutoSpillStatus>>>(partition_num);
     for (auto & memory : *partition_revocable_memories)
         memory = 0;
-    for (auto & status : *partition_is_spilled)
-        status = false;
+    for (auto & is_spilled : *partition_is_spilled)
+        is_spilled = false;
+    for (auto & spill_status : *partition_spill_status)
+        spill_status = AutoSpillStatus::NO_NEED_AUTO_SPILL;
 }
 
 Int64 HashJoinSpillContext::getTotalRevocableMemoryImpl()
 {
     Int64 ret = 0;
-    for (const auto & x : *partition_revocable_memories)
-        ret += x;
+    for (size_t part_index = 0; part_index < partition_revocable_memories->size(); ++part_index)
+    {
+        if (in_spillable_stage || isPartitionSpilled(part_index))
```

Comment:
```suggestion
        if (isPartitionSpilled(part_index))
```

Reply:
Actually, `in_spillable_stage` is the build stage for HashJoin, if some partition is spilled, during probe, it still need spill. It's not a very suitable name, I will refine it for better understanding